tests/src/Integration/Action/SystemPageRedirectTest.php

```diff
@@ -0,0 +1,182 @@
+     */
+    protected $logger;
+
+    /**
+     * The mocked logger factory service.
+     *
+     * @var \Drupal\Core\Logger\LoggerChannelFactoryInterface|\Prophecy\Prophecy\ProphecyInterface
+     */
+    protected $loggerFactory;
+
+    /**
+     * The mocked request stack service.
+     *
+     * @var \Symfony\Component\HttpFoundation\RequestStack|\Prophecy\Prophecy\ProphecyInterface
+     */
+    protected $requestStack;
+
+    /**
+     * The mocked current path stack service.
+     *
+     * @var \Drupal\Core\Path\CurrentPathStack|\Prophecy\Prophecy\ProphecyInterface
+     */
+    protected $currentPathStack;
+
+    /**
+     * A mocked request.
+     *
+     * @var \Symfony\Component\HttpFoundation\Request|\Prophecy\Prophecy\ProphecyInterface
+     */
+    protected $currentRequest;
+
+    /**
+     * A mocked parameter bag.
+     *
+     * @var \Symfony\Component\HttpFoundation\ParameterBag|\Prophecy\Prophecy\ProphecyInterface
+     */
+    protected $parameterBag;
+
+    /**
+     * The action to be tested.
+     *
+     * @var \Drupal\rules\Plugin\RulesAction\SystemPageRedirect
+     */
+    protected $action;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setUp() {
+      parent::setUp();
+
+      // Mock a logger.
+      $this->logger = $this->prophesize(LoggerInterface::class);
+
+      // Mock the logger service, make it return our mocked logger, and register
+      // it in the container.
+      $this->loggerFactory = $this->prophesize(LoggerChannelFactoryInterface::class);
+      $this->loggerFactory->get('rules')->willReturn($this->logger->reveal());
+      $this->container->set('logger.factory', $this->loggerFactory->reveal());
+
+      // Mock a parameter bag.
+      $this->parameterBag = $this->prophesize(ParameterBag::class);
+
+      // Mock a request, and set our mocked parameter bag as it attributes
+      //property.
+      $this->currentRequest = $this->prophesize(Request::class);
+      $this->currentRequest->attributes = $this->parameterBag->reveal();
+
+      // Mock the request stack, make it return our mocked request when the
+      // current request is requested, and register it in the container.
+      $this->requestStack = $this->prophesize(RequestStack::class);
+      $this->requestStack->getCurrentRequest()->willReturn($this->currentRequest);
+      $this->container->set('request_stack', $this->requestStack->reveal());
+
+      // Mock the current path stack.
+      $this->currentPathStack = $this->prophesize(CurrentPathStack::class);
+      $this->container->set('path.current', $this->currentPathStack->reveal());
+
+      // Instantiate the redirect action.
+      $this->action = $this->actionManager->createInstance('rules_page_redirect');
+    }
+
+    /**
+     * Tests redirection.
+     *
+     * @covers ::execute
+     */
+    public function testRedirect() {
+      $this->currentPathStack->getPath()->willReturn('/some/random/test/path');
```

Comment:
The docs on the the getPath() method say they return a path without leading slash, so we should do that as well.